orchestrator/orchestrator.go

```diff
@@ -12,8 +14,8 @@ type Orchestrator interface {
 	// Logger gets a logger.
 	Logger() log.Logger
 
-	// SessionProvider gets a session provider.
-	SessionProvider() session.Provider
+	// SessionProvider gets a session for the provided request.
+	SessionProvider(*http.Request) (session.Provider, error)
```

Comment:
Could we add the response here, too? This would be critical for supporting a cookie store, etc.

Reply:
Thanks for bringing that up Granville. I'd like to better understand that use case actually. I was trying to think of it in terms of implementing it in a service extension and when it would make sense. 

Basically, I'm trying to avoid a situation where we have one of these [in our docs](https://docs.strata.io/orchestrator-reference/app-gateway/service-extension-authorization#isauthorized-service-extension):
> Although this Service Extension is being passed the http.ResponseWriter, typically it will not send any response directly to the user. Writing to the http.ResponseWriter's body or headers may produce undesired behavior.

Would it make sense to pass this in as a variadic arg in the future in case we are worried about changing signatures?

Reply:
Other questions I had in mind: What if the Service Extension doesn't have a http.ResponseWriter to pass to this method?

Reply:
Apologies for the late ping here Granville but I'm happy to discuss this more sometime this week if possible. I'll be OOO for most of tomorrow as I'll be traveling to YVR but happy to keep this conversation going async.

Reply:
I'll start putting a commit together to include the ResponseWriter in the func signature because I'm making the assumption that it is needed for the Orchestrator Groups session routing feature.